# Galeano-Nino-Bullman-Intratumoral-Microbiota-2022


[![DOI](https://zenodo.org/badge/530442339.svg)](https://zenodo.org/badge/latestdoi/530442339)


Analysis code used in Galeano Nino et al., Impact of Intratumoral Microbiota on Spatial and Cellular Heterogeneity in cancer. 2022

The code in this repository is organized to reflect the description in the Methods
section of Galeano Nino et al., Impact of Intratumoral Microbiota on Spatial and Cellular Heterogeneity in cancer. 2022.

## Environment and Reference Data

### Environment

All of the analysis code documented in this repository was run on the shared computing cluster
maintained at the Fred Hutchinson Cancer Research Center between May 2020 and August 2022.
The software dependencies used by these scripts are provided using the EasyBuild installation
maintained by the Fred Hutch Scientific Computing group.
Those software dependencies are loaded into the environment with the `ml` command (e.g. `ml CellRanger/6.1.1`).

### Reference Data

Prior to running the analysis scripts, reference databases were downloaded for PathSeq (December 2020)
and CellRanger (January 2022).
The location of those reference databases is provided to the analysis scripts using the environment variables `PATHSEQ_DB` and `CELLRANGER_DB`.

# Overview of the Computational Pipeline for INVADEseq
## Part 1: 10x Single cell data (For cell culture samples and patient samples)
###   Input Data:
   - All of the input data for this analysis is provided in FASTQ format generated by the CellRanger `mkfastq` command
   - The folder containing those FASTQ files is set to the environment variable `CELLRANGER_MKFASTQ_FOLDER`
###   Preprocess:
   1. Identification of microbial reads within single cells GEX libraries (`Pipeline_GEX.sh`)
   2. INVADEseq bacterial 16S rRNA gene libraries (`Pipeline_16S.sh`). The variable `CELLRANGER_FOLDER_NOVA` should be set to the output folder from the `Pipeline_GEX.sh` script.
   3. Combine and deduplication of microbial metadata from step 1 & 2 (`merge_metadata.py` and `metadata_dedup.py`). The folder used as outputs from the previous steps should be provided as an argument to the `merge_metadata.py` script.
###   Output Data:
   - `headneck_GEX_16s_mix_dedup.csv` contains bacteria UMI counting matrix that can be used as metadata in single cell process.
###   Processing of single cell data
   1. Seurat data processing, Harmony integration, SingleR annotation and copyKAT predication (`Single_Cell.r`)
   2. Differentially expression analysis and GSEA (`DE.r`)
## Part 2: 10x Visium spatial transcriptomic data
   1. Identification of microbial reads within 10x Visium spatial transcriptomic data (`Pipeline_Visium.sh`)
   2. Bioinformatic analysis of 10x Visium spatial transcriptomic data (`Visium.r`)
   3. summarize numbers of bacteria reads and UMIs in 10X Visium data (`validate_and_count.py`) The folder used as outputs from the previous steps should be provided as an argument to the `Pipeline_Visium.sh` script.
###   Output Data:
   - `CRC_16.visium.raw_matrix.genus.csv` and `OSCC_2.visium.raw_matrix.genus.csv` contain bacteria UMI counting matrix that can be used as metadata in visium data process
   - `CRC_16.visium.raw_matrix.validate.csv` and  `OSCC_2.visium.raw_matrix.validate.csv` contain validation data that can be used as the input of `validate_and_count.py`
